dashboard/app/controllers/maker_controller.rb

```diff
@@ -5,5 +5,9 @@ def setup
   end
 
   def discountcode
+    return head :not_found unless current_user.teacher?
```

Comment:
This could potentially be handled by `CanCan`, adding an entry to ability.rb and using `authorize_resource` instead of `authenticate_user!` in this controller